src/Sarif/NotYetAutogenerated/SuppressionStates.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.CodeDom.Compiler;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// The state of a result relative to a baseline of a previous run.
+    /// </summary>
+    [GeneratedCode("Microsoft.Json.Schema.ToDotNet", "0.16.0.0")]
+    [Flags]
+    public enum SuppressionStates
+    {
+        None,
```

Comment:
If we have a value of Unknown, then we need an explicit value for NotSuppressedInSource. Now it becomes unclear how to manage this as a [Flags] enum. We would have Unknown, SuppressedInSource, NotSuppressedInSource as a non-flags enum and SuppressedInBaseline would be handled elsewhere. With the current approach, we maintain a set of exclusion conditions in the event that we have some positive signal of this kind. I think this is a better approach but am open to continued discussion.